client/components/status/status.js

```diff
@@ -1,55 +1,191 @@
-      },
-      url,
-      data: qs.stringify({
-        token,
-      }),
-    })
-      .then(response => {
-        if (response.data["control:Auth-Type"] !== "Accept") {
+  constructor(props) {
+    super(props);
+    this.loginIfameRef = React.createRef();
+    this.loginFormRef = React.createRef();
+    this.logoutIfameRef = React.createRef();
+    this.logoutFormRef = React.createRef();
+    this.state = {
+      username: "",
+      password: "",
+      sessions: [],
+    };
+    this.validateToken = this.validateToken.bind(this);
+    this.getUserRadiusSessions = this.getUserRadiusSessions.bind(this);
+  }
+
+  async componentDidMount() {
+    const {cookies, orgSlug} = this.props;
+    // to prevent recursive call in case redirect url is status page
+    if (window.top === window.self) {
+      try {
+        const {location, captivePortalLoginForm} = this.props;
+        const searchParams = new URLSearchParams(location.search);
+        const macaddr = searchParams.get(
+          captivePortalLoginForm.macaddr_param_name,
+        );
+        if (macaddr) {
+          cookies.set(`${orgSlug}_macaddr`, macaddr, {path: "/"});
+        }
+      } catch {
+        //
+      }
+      const isValid = await this.validateToken();
+      if (isValid) {
+        const macaddr = cookies.get(`${orgSlug}_macaddr`);
+        if (macaddr) {
+          await this.getUserRadiusSessions();
+          /* request to captive portal is made only if there is
+            no active session from macaddr stored in the cookie */
+          const {sessions} = this.state;
+          if (sessions && sessions.length === 0) {
+            if (this.loginFormRef && this.loginFormRef.current)
+              this.loginFormRef.current.submit();
+          }
+        } else if (this.loginFormRef && this.loginFormRef.current)
+          this.loginFormRef.current.submit();
+      }
+    }
+  }
+
+  async getUserRadiusSessions() {
+    const {cookies, orgSlug, logout} = this.props;
+    const url = getUserRadiusSessionsUrl(orgSlug);
+    try {
+      const response = await axios({
+        method: "get",
+        headers: {
+          "content-type": "application/x-www-form-urlencoded",
+        },
+        url,
+      });
+      this.setState({sessions: response.data});
+    } catch (error) {
+      logout(cookies, orgSlug);
+      toast.error(genericError, {
+        onOpen: () => toast.dismiss(mainToastId),
+      });
+      logError(error, genericError);
+    }
+  }
+
+  handleLogout = async () => {
+    const {setLoading} = this.context;
+    const {orgSlug, logout, cookies} = this.props;
+    setLoading(true);
+    await this.getUserRadiusSessions();
+    const {sessions} = this.state;
+    if (sessions.length > 0) {
+      if (this.logoutFormRef && this.logoutFormRef.current) {
+        this.logoutFormRef.current.submit();
+        setTimeout(() => {}, 0);
+      }
+    }
+    logout(cookies, orgSlug);
+    setLoading(false);
+    toast.success(logoutSuccess);
```

Comment:
if I comment out these 3 lines (106, 107 and 108), the captive portal logout works.
I think these lines should be executed only once `this.logoutFormRef.current.submit()` has finished, otherwise it seems the submit is not triggered.
Can you add listener to the onload of the iframe or anything similar so we can execute these 3 lines only after the captive portal logout has completed?

Keep in mind the same should happen for the login as well, if the captive portal login has to be executed, the success message should be shown only once the captive portal has replied, otherwise users may try to visit another page and be redirected again to the captive page and then they'll be super confused.

With this solved we're ready to merge and move on! :+1: :clap: